app/controllers/reservations_controller.rb

```diff
@@ -31,6 +31,8 @@ def edit
   end
 
   def update
+    params[:credit_card_id] = @reservation.credit_card_id
+    set_credit_card
```

Comment:
can also just make this one line as @credit_card = CreditCard.find(@reservation.credit_card_id) if we want.